### PR TITLE
Updates Stylelint to 15.0.0

### PR DESCRIPTION
**Pull request by Antonio-Laguna**
Related Issue/RFC: #284 

### Description of the Change

This updates `stylelint` to version `15.0.0` and decouple from `@wordpress/stylelint-config`. Upgrading to this should be painless. 

Expanding a bit:

Stylelint 15 soft deprecates 76 stylistic rules of which we were leveraging 51 rules. Right now, any effort on `@wordpress/stylelint-config` is halted given they'll need to discuss what to do with those rules so forking from them seemed the shortest path that allows us to move forward.

This has been discussed on https://github.com/10up/10up-toolkit/issues/295

### Checklist:

- [X] I have read the [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document.
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.
- [X] I have added a changeset to my PR. See [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document for instructions

### CODE REVIEW

**Comment by Antonio-Laguna** [comment]
@nicholasio after the upgrade I can go and check the lining once we upgrade the packages

**Comment by nicholasio** [comment]
Any idea why the test failed on node 16?

**Comment by Antonio-Laguna** [comment]
@nicholasio failed on linting. Only lints on Node 16. Checking the issue, what fails linting is stylelint for the 10up-theme. I think after things are published I can go and update everything there?

**Comment by nicholasio** [comment]
The failure is legit and not specific to node 16

image

We only run linting in one of the jobs and apparently, our deps update is forcing everything to be react 18 now.

I'm wondering if we should just make this a warning instead of an error.

File: packages/eslint-config/package.json [hunk 1]
```diff
@@ -50,7 +51,7 @@
 	"peerDependencies": {
 		"@babel/core": "^7.21.4",
 		"@babel/eslint-parser": "^7.21.3",
-		"@wordpress/eslint-plugin": "^13.10.0",
+		"@wordpress/eslint-plugin": "^14.9.0",
```
Comment (nicholasio) [inline packages/eslint-config/package.json:54]: I believe this is forcing react 18

**Comment by Antonio-Laguna** [comment]
These new errors aren't what they used to be! See https://github.com/10up/10up-toolkit/actions/runs/5738199032/job/15551300090#step:8:76

That said, I'm wondering if we should upgrade to React 18

File: projects/10up-theme/assets/css/frontend/testing.scss [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // Test comment
-@import "./components/sass-component";
+@import url("./components/sass-component.scss");
```
Comment (Antonio-Laguna) [inline projects/10up-theme/assets/css/frontend/testing.scss:2]: @nicholasio thoughts here? I think being consistent between `scss` and no `scss` is the way to go. Doesn't seem possible to have one linting rule _just_ for `scss` rules so it's one or the other.

Comment (nicholasio) [reply projects/10up-theme/assets/css/frontend/testing.scss:2]: fine by me!

**Review by nicholasio — DISMISSED** [review]

**Review by nicholasio — APPROVED** [review]

**Pull request merged**